examples/todo/src/__tests__/integration/services/geocoder.service.integration.ts

```diff
@@ -23,15 +25,16 @@ describe('GeoLookupService', function() {
   let service: GeocoderService;
   before(givenGeoService);
 
+  let available = true;
+  before(async () => {
+    available = await isGeoCoderServiceAvailable(service);
+  });
+
   it('resolves an address to a geo point', async () => {
-    const points = await service.geocode('1 New Orchard Road, Armonk, 10504');
-
-    expect(points).to.deepEqual([
-      {
-        y: 41.109653,
-        x: -73.72467,
-      },
-    ]);
+    if (!available) return;
+    const points = await service.geocode(aLocation.address);
```

Comment:
The geocode for a certain address is already computed when checking the availability, it would be better to only run `service.geocode()` once, like returning the result OR true/false/error so we don't need to do it again here.

Reply:
It should be the responsibility of http-caching-proxy.